lib/charms/prometheus_k8s/v0/prometheus_remote_write.py

```diff
@@ -0,0 +1,124 @@
+# Copyright 2021 Canonical Ltd.
+# See LICENSE file for licensing details.
+"""This library facilitates the prometheus_remote_write interface.
+
+For the provider charm no communication is necesssary. For the consumer charm,
+instantite a PrometheusRemoteWriteConsumer in the __init__ method of your charm
+class:
+
+```
+from charms.prometheus_k8s.v0.prometheus_remote_write import PrometheusRemoteWriteConsumer
+
+def __init__(self, *args):
+    ...
+    self.remote_write = PrometheusRemoteWriteConsumer(self, "remote-write")
+    ...
+```
+
+Then access the endpoints with:
+`self.remote_write.endpoints`
+or access a prometheus style config object with:
+`self.remote_write.configs`
+"""
+
+from typing import List, Optional, Union
+
+from ops.charm import CharmBase
+from ops.framework import Object
+from ops.model import BlockedStatus
+
+
+class PrometheusRemoteWriteConsumer(Object):
+    """A prometheus remote write consumer."""
+
+    def __init__(self, charm: CharmBase, relation_name: str):
+        """A prometheus remote write consumer.
+
+        Args:
+            charm: The charm object that instantiated this class.
+            relation_name: The relation name as defined in metadata.yaml.
+        """
+        super().__init__(charm, relation_name)
+        self._relation_name = relation_name
```

Comment:
I am a bit surprised that this charm library does not respond to any events. Is that because it only supports interaction with a single remote write endpoint ? What if no remote write endpoint is available but becomes available at a later point in time. In this case it it worth emitting a event to notify the consumer charm that a remote write endpoint has become available ? I suppose this question is related to the question in docstring above on how to use the `endpoints()` and `config()` method.

Reply:
So in the past we have written libraries to always be the object that listens to `relation_changed` even if we were not doing anything with it. This is something that was more or less dictated to us a long time ago. I think, however that it does not always make sense.

Specifically regarding this charm and the comment above. If new endpoints become available, a `relation_changed` event will trigger. One of two things can happen. First, the charm itself could listen to relation changed and do whatever it needs to do. Second, we could listen here, do nothing, and emit a different custom event for the charm to respond to. To me the former seems a lot more simple and transparent.

There are definitely cases when custom events are appropriate, I think, however, that in this case it would be no more than a rename of the already aptly named `relation_changed`.